script.js

```diff
@@ -1 +1,129 @@
+        if (isDuplicate) {
+            alert("이미 동일한 투두가 있습니다!👏🏻"); // 알림창으로 중복 투두 알림
+            todoInput.value = "";
+            return; // 중복되면 함수 종료
+        }
+
+      todoList.push({ text: newTodo, completed: false }); // 배열에 입력 값 저장 
+      saveStorage(); // list에 새로운 todo가 추가 됨으로써 변경되었으니 다시 localStorage에 todoList 저장
+      displayTodo(newTodo, false); 
+      todoInput.value = ""; // 배열에 todo를 저장하고 렌더링 했다면 input을 지워서 다시 입력할 수 있도록
+      updateCounts(); // 전체 todo 개수에 +1
+  }
+
+  function saveStorage() {
+      localStorage.setItem("todos", JSON.stringify(todoList));  //localStorage는 문자열 형식의 데이터만 저장할 수 있기 때문에, JSON.stringify()를 사용해 자바스크립트 객체나 배열을 JSON 문자열로 변환한 후 저장. 다시 불러올 때는 JSON.parse() 이용
+  }  // setItem(key,value) 특정 key에 해당 value 할당 
+
+  function loadStorage() {
+      const storedTodos = localStorage.getItem("todos"); // 기존에 localStorage에 저장되어있던 배열을 불러오기, 만약 없다면 null이 저장 됨 
+      if (storedTodos) {
+          todoList = JSON.parse(storedTodos);
+          todoList.forEach((todo) => displayTodo(todo.text, todo.completed)); // todoList 배열을 순회하며 저장된 모든 todo를 화면에 렌더링
+      }
+      updateCounts(); // 새로 고침 시 기존 todoList 배열 불러와서 총 개수 맞게 렌더링
+  }
+
+  function displayTodo(todoText, isCompleted) {
+      const li = document.createElement("li"); // 새로운 <li>요소 생성. 하나의 todo를 나타냄
+
+      const checkbox = document.createElement("input"); // 새로운 input 요소 checkbox 생성
+      checkbox.type = "checkbox"; // 이 요소의 type = checkbox
+      checkbox.classList.add("todoCheckBox"); // 스타일링을 위해 클래스 목록에 클래스 이름 추가! 즉 checkbox에 할당되는 클래스 이름이 todo-checkbox
+      checkbox.checked = isCompleted; // isCompleted가 true라면 checkbox가 체크 됨
+
+      checkbox.addEventListener("change", function () { // checkbox의 상태가 바뀔 때, 즉 checked의 속성이 변경될 때 (사용자가 체크박스를 체크하거나 해제할 때) 실행될 이벤트 핸들러
+          toggleTodoCompletion(todoText);
+          li.querySelector("span").style.textDecoration = checkbox.checked ? "line-through" : "none";
+      }); // <li> 요소의 첫 번째 <span> 요소를 찾아 스타일링. 만약 체크박스가 체크 되어있으면? 선을 긋고 해제되어 있으면 선을 없앤다!
+
+      const todoSpan = document.createElement("span"); //<li> 요소 안에 쓰여질 텍스트를 <span> 요소로 설정
+      todoSpan.classList.add("todoSpan");
+      todoSpan.textContent = todoText; // 매개변수로 전달 받은 input의 text를 변수 todoSpan의 텍스트 내용으로 저장
+      if (isCompleted) {
+          todoSpan.style.textDecoration = "line-through";
+      }
+
+      const deleteBtn = document.createElement("button"); // 새로운 <button> 요소 생성
+      deleteBtn.textContent = "삭제"; // deleteBtn의 텍스트 내용을 삭제라고 지정
+      deleteBtn.classList.add("deleteBtn");
+      deleteBtn.addEventListener("click", function () { // 버튼이 클릭 되었을 때 실행될 함수!!
+          alert('정말 삭제하시겠습니까?');
+          deleteTodo(todoText, li);
+      });
+
+      li.appendChild(checkbox); // 새로 만든 요소 (체크박스, 텍스트, 삭제 버튼)을 <li> 요소에 추가해서 하나의 todo 항목 완성!
+      li.appendChild(todoSpan); // 할 일 텍스트를 담은 <span> 요소 추가
+      li.appendChild(deleteBtn); // 삭제 버튼 추가
```

Comment:
`appendChild`는 하나의 요소만 자식요소로 넣을 수 있다고 해요!
만약 여러 요소를 하나의 태그 안에 자식 요소로 넣고 싶다면
`append`를 이용하는 것도 좋은 방법일 것 같아요! 보다 코드가 간소화되니까요!
```suggestion
      li.append(checkbox, todoSpan, deleteBtn) 
```

Reply:
헉... 저도 이거 똑같은 내용을 썼답니다! 물론 제가 수정해야 한다는 쪽으로요... ㅎㅎ 코드 리뷰를 통해 하나 더 배우게 되어 기쁘네요 감사합니다 🔥🔥